test/unit/test_amqpclient.js

```diff
@@ -106,7 +106,7 @@ describe('AMQPClient', function() {
 
       return client.connect(mock_uri)
         .then(function () {
-          return client.createSender(queue, { name: 'queue_TX' });
+          return client.createSender(queue, { options: { name: 'queue_TX' } });
```

Comment:
Honestly, I'd be ok with special-casing name. It does feel like a) something people want to commonly set, and b) something we use internally far more than just in attaching.